### ISSUE TITLE
Support additional JWT algorithms

### ISSUE THREAD

**Issue by kevinbader** [Accenture/reactive-interaction-gateway#84]
# Currently

RIG [supports `hs256` only](https://github.com/Accenture/reactive-interaction-gateway/blob/6564fd2e9753cd5cfdb4dea561e942c68db773ef/apps/rig_auth/lib/rig_auth/jwt/utils.ex#L31); the secret is [read from `JWT_SECRET_KEY`](https://github.com/Accenture/reactive-interaction-gateway/blob/6564fd2e9753cd5cfdb4dea561e942c68db773ef/docs/rig-ops-guide.md#L22).

# Idea:

Joken, the library we're using for handling JWTs, [supports more than just the HS* algorithms](https://hexdocs.pm/joken/Joken.Signer.html). The idea is to use the `hs`/`rs`/... functions found in [`signer.ex`](https://github.com/bryanjos/joken/blob/master/lib/joken/signer.ex) and support specifying the algorithm using an additional environment variable. For example:

```bash
JWT_ALG=HS256
JWT_KEY=my-symmetric-key
```

or

```bash
JWT_ALG=RS256
JWT_KEY=my-public-key
```

Basically, you'd do something along the lines of:

```elixir
jwt_alg
|> String.downcase
|> case do
  "hs" <> _ = alg -> Signer.hs(alg, jwt_key)
  "rs" <> _ = alg -> Signer.rs(alg, jwt_key)
  ...
end
```

# Additional thoughts

Joken does not support token _encryption_, which might be a future requirement. [Here's a nice overview about JWT algorithms and what they're used for](https://connect2id.com/products/nimbus-jose-jwt/algorithm-selection-guide).
